### PR TITLE
Fix v-image's intersection observer sometimes preventing image load

### LINKED ISSUE #15078: File library images disappear on sorting
### Describe the Bug

When you initially the File Library in Directus shows all our images from a Google Cloud bucket correctly. However, when you click the sort button the images randomly disappear. If you keep clicking for about 4-5 times suddenly everything is complete again.  

image

I tested this with images in the local storage and that seemed to work fine, I didn't have any issues. I also tested it with Firefox, no issues there at all, both cloud based images and local images are working fine. So it seems to be mainly a Chrome issue. 
Rijk suggested it might be the intersection observer, so I checked what happened if you resize the screen so the images go out of view and back in and this also makes the images reappear. 

This issue is also present in the Users section, when sorting users the avatars disappear / reappear in the same way.

### To Reproduce

Upload 10 images in a GCS bucket
View them from Directus in the File Library
Press the sort button a couple of times

https://user-images.githubusercontent.com/108686553/184691536-4253e8c9-5eb7-4125-a354-387ea681b605.mp4

### Errors Shown

No errors in the console, the network inspector also shows all the images in the API calls. 

### What version of Directus are you using?

9.15.1

### What version of Node.js are you using?

16.15

### What database are you using?

MySQL

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

### PR DESCRIPTION
**Pull request by azrikahar**
## Description

Fixes #15078

As reported by the user, there seems to be a difference between Chrome and Firefox (untested on Safari) on intersection observers.

Currently `<v-image>` uses the `inView` ref to determine whether to "show" an image or not, and it gets it's value from the FIRST entry:

https://github.com/directus/directus/blob/14f0acbcc605cb403b7cd5aebd480953d68c5631/app/src/components/v-image.vue#L30

However, in Chrome, there can be up to 2 intersectionObserverEntry when we are flipping the sort, hence the logic of using `entries[0]` falters since it may be `entries[1]` that correctly returns true for `isIntersecting`:

https://user-images.githubusercontent.com/42867097/184801832-45d1c614-b039-4ead-9313-4dcd74e98002.mp4

As we can see in Firefox, it will always just be 1 intersectionObserverEntry, hence it always works:

https://user-images.githubusercontent.com/42867097/184801896-68f445f6-385d-4bba-9704-242bfb6a23c8.mp4

I've also tried ensuring `observer.observe()` isn't getting fired twice, but seems like Chrome still consistently has up to 2 instances of intersectionObserverEntry per element. 

Based on the following quote:

> When the callback is invoked, it receives a list of IntersectionObserverEntry objects, one for each observed target which has had the degree to which it intersects the root change such that the amount exposed crosses over one of the thresholds, in either direction.
> (from https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)

I assume it's some form of optimizations that "hijacks" the list of thresholds, which technically still falls under the spec, or it's an actual bug in Chrome.

For now, this PR resolves this by looping through the entries instead. The `loaded` flag should prevent multi loads, but even when there's 2 IntersectionObserverEntry, one of them is always not intersecting so it's still one load attempt.

## Type of Change

- [x] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated

### CODE REVIEW

File: app/src/components/v-image.vue [hunk 1]
```diff
@@ -25,13 +25,16 @@ const srcData = ref<string>(emptyPixel);
...
-const observer = new IntersectionObserver(async (entries) => {
+const observer = new IntersectionObserver((entries) => {
 	if (entries.length === 0) return;
-	inView.value = entries[0].isIntersecting;
 
-	if (entries[0].isIntersecting && !loaded && props.src) {
-		loadImage();
-	}
+	entries.forEach((entry) => {
+		inView.value = entry.isIntersecting;
```
Comment (rijkvanzanten) [inline app/src/components/v-image.vue:32]: Is there ever a case where the second entry doesn't equal the first entry? If so, we might have to make this a little smarter, as the second entry can override the state of the first entry

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/components/v-image.vue (hunk 1 above)
Comment (azrikahar) [reply app/src/components/v-image.vue:32]: Yeah that's a great point. They are definitely not equal, with one of them being the chrome-specific one where it has 0 intersecting value and no root bounds:

chrome_4hZZGAhFJA

Thoughts on using the following logic with `.some()` instead? That way there shouldn't be overriding as long as one of them is true. Tested it and it does work:

```js
	const isIntesecting = entries.some((entry) => entry.isIntersecting);
	inView.value = isIntesecting;

	if (isIntesecting && !loaded && props.src) {
		loadImage();
	}
```

Comment (rijkvanzanten) [reply app/src/components/v-image.vue:32]: @azrikahar That first one is confusing. It's the same `img.image`, but with 0x0 dimensions? Can you confirm that's the same element reference?

Comment (azrikahar) [reply app/src/components/v-image.vue:32]: Yeah somehow it is the same element, down to the exact same base64 `src`, `alt` etc. They even have the same `x`, `y`, `height`, `width` 🤔 but oddly the IntersectionObserverEntry properties can have this disparity.

Comment (rijkvanzanten) [reply app/src/components/v-image.vue:32]: @azrikahar Is the "correct" one always the second/last one?

Comment (azrikahar) [reply app/src/components/v-image.vue:32]: Ya seems like it is from what I observed 👍

image

Comment (rijkvanzanten) [reply app/src/components/v-image.vue:32]: OK! In that case, should we start with an `entries.at(-1)` instead of looping over them?

Comment (azrikahar) [reply app/src/components/v-image.vue]: Sounds good! Tested and works well. Pushed the change 👍

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**